### PR TITLE
Redesign last step in email creation [MAILPOET-3089] [MAILPOET-2790]

### PR DESCRIPTION
**Pull request by NeosinneR**
[MAILPOET-3089] [MAILPOET-2790]

[MAILPOET-3089]: https://mailpoet.atlassian.net/browse/MAILPOET-3089
[MAILPOET-2790]: https://mailpoet.atlassian.net/browse/MAILPOET-2790

### CODE REVIEW

**Review by wxa — CHANGES_REQUESTED** [review]
The biggest issue is validation failures do not stop a form from saving, allowing to save an email without subject/list specified, or causing SQL errors due to very long data etc.

File: assets/js/src/newsletters/automatic_emails.jsx [hunk 1]
```diff
@@ -48,14 +48,34 @@ const configureSendPageOptions = (defaultFields, newsletter) => {
...
+      fields: [
+        {
+          name: 'subject',
+          customLabel: MailPoet.I18n.t('subjectLabel'),
+          class: 'mailpoet-form-field-subject',
+          placeholder: MailPoet.I18n.t('subjectLine'),
+          tooltip: MailPoet.I18n.t('subjectLineTip'),
+          type: 'text',
+          validation: {
+            'data-parsley-required': true,
```
[the remark this answers is not in the document]
Comment (NeosinneR) [reply assets/js/src/newsletters/automatic_emails.jsx:63]: Fixed, added regular HTML `maxlength` attribute, which won't allow entering any more characters

[the remark this answers is not in the document]
Comment (wxa) [reply assets/js/src/newsletters/automatic_emails.jsx:63]: You've fixed it for preheader, but not for subject.

[the remark this answers is not in the document]
Comment (NeosinneR) [reply assets/js/src/newsletters/automatic_emails.jsx:63]: Yeah, sorry, completely missed that you were talking about both preview and subject. Now, the subject also has `maxlength` added.

File: assets/js/src/newsletters/automatic_emails.jsx [hunk 2]
```diff
@@ -48,14 +48,34 @@ const configureSendPageOptions = (defaultFields, newsletter) => {
...
+        },
+        {
+          name: 'preheader',
+          customLabel: MailPoet.I18n.t('preheaderLabel'),
+          class: 'mailpoet-form-field-preheader',
+          placeholder: MailPoet.I18n.t('preheaderLine'),
+          tooltip: `${MailPoet.I18n.t('preheaderLineTip1')} ${MailPoet.I18n.t('preheaderLineTip2')}`,
+          type: 'textarea',
+          validation: {
+            'data-parsley-maxlength': 250,
```
[the remark this answers is not in the document]
Comment (NeosinneR) [reply assets/js/src/newsletters/automatic_emails.jsx]: That's by design. The two lines are enough for the max 250 characters. I haven't found any research on using new lines, but in a [SO](https://stackoverflow.com/questions/47166603/why-is-there-a-line-break-in-my-email-preview-text-in-ios-mail-app) post the multiline preview looks fine in the iOS Mail app. I however added a note in the tooltip (proofread by Laura), that we recommend using it on a single line and we'll see in the future if we receive any issues regarding this.

**Comment by wxa** [comment]
Subject validation error message is partially obstructed when the preheader field is highlighted in green or red:
изображение
Same with the sender name field:
изображение

**Review by wxa — APPROVED** [review]
Huge work, thanks!

**Review by veljkho — APPROVED** [review]

**Pull request merged**